### PR TITLE
Add check for invalid date range on storage load

### LINKED ISSUE #332: Edited json file
Edited the json file and end date occurs before start. 

image

### PR DESCRIPTION
**Pull request by eeliana**
- fix #332

### CODE REVIEW

**Review by Chesterwongz — COMMENTED** [review]
Could you add all the checks to the test

File: src/main/java/seedu/address/storage/JsonAdaptedLesson.java [hunk 1]
```diff
@@ -101,6 +103,9 @@ public Lesson toModelType() throws IllegalValueException {
         }
         final Date modelDate = new Date(StringUtil.stripLeadingZeroes(strippedDate));
         final Date modelEndDate = new Date(StringUtil.stripLeadingZeroes(strippedEndDate));
+        if (modelDate.isAfter(modelEndDate)) {
```
Comment (Chesterwongz) [inline src/main/java/seedu/address/storage/JsonAdaptedLesson.java:106]: Is end date same as date meant to be valid?

Comment (eeliana) [reply src/main/java/seedu/address/storage/JsonAdaptedLesson.java:106]: Yes, you can start and end the lesson on the same date.

Comment (eeliana) [reply src/main/java/seedu/address/storage/JsonAdaptedLesson.java:106]: Behaviour is consistent with calendar apps out there, but true that it doesn't make sense to have a single date recurring lesson. I don't think it's an issue though

**Review by Chesterwongz — APPROVED** [review]
LGTM

**Pull request merged**